models/iaf_cond_alpha_mc_kinetics.cpp

```diff
@@ -0,0 +1,875 @@
+  E_in[ DIST ] = -85.0;   // mV
+  E_L[ DIST ] = -70.0;    // mV
+  tau_synE[ DIST ] = 0.5; // ms
+  tau_synI[ DIST ] = 2.0; // ms
+  I_e[ DIST ] = 0.0;      // pA
+  // changes
+  tau_curAP[ DIST ] = 1.0; // ms
+  amp_curAP[ DIST ] = 0.0; // pA
+}
+
+nest::iaf_cond_alpha_mc_kinetics::Parameters_::Parameters_( const Parameters_& p )
+  : V_th( p.V_th )
+  , V_reset( p.V_reset )
+  , t_ref( p.t_ref )
+  ,
+  // changes
+  V_max( p.V_max )
+  , E_Ca( p.E_Ca )
+  , G_Ca( p.G_Ca )
+  , tau_m( p.tau_m )
+  , tau_h( p.tau_h )
+  , half_m( p.half_m )
+  , half_h( p.half_h )
+  , slope_m( p.slope_m )
+  , slope_h( p.slope_h )
+  , jump_Th( p.jump_Th )
+  , tau_Th( p.tau_Th )
+  , act_flag( p.act_flag )
+  , reset_flag( p.reset_flag )
+
+{
+  // copy C-arrays
+  for ( size_t n = 0; n < NCOMP - 1; ++n )
+    g_conn[ n ] = p.g_conn[ n ];
+
+  for ( size_t n = 0; n < NCOMP; ++n )
+  {
+    t_L[ n ] = p.t_L[ n ];
+    nt_L[ n ] = p.nt_L[ n ];
+    g_L[ n ] = p.g_L[ n ];
+    C_m[ n ] = p.C_m[ n ];
+    E_ex[ n ] = p.E_ex[ n ];
+    E_in[ n ] = p.E_in[ n ];
+    E_L[ n ] = p.E_L[ n ];
+    tau_synE[ n ] = p.tau_synE[ n ];
+    tau_synI[ n ] = p.tau_synI[ n ];
+    I_e[ n ] = p.I_e[ n ];
+    // changes
+    tau_curAP[ n ] = p.tau_curAP[ n ];
+    amp_curAP[ n ] = p.amp_curAP[ n ];
+  }
+}
+
+nest::iaf_cond_alpha_mc_kinetics::Parameters_& nest::iaf_cond_alpha_mc_kinetics::Parameters_::
+operator=( const Parameters_& p )
+{
+  assert( this != &p ); // would be bad logical error in program
+
+  V_th = p.V_th;
+  V_reset = p.V_reset;
+  t_ref = p.t_ref;
+  // changes
+  V_max = p.V_max;
+  E_Ca = p.E_Ca;
+  G_Ca = p.G_Ca;
+  tau_m = p.tau_m;
+  tau_h = p.tau_h;
+  half_m = p.half_m;
+  half_h = p.half_h;
+  slope_m = p.slope_m;
+  slope_h = p.slope_h;
+  jump_Th = p.jump_Th;
+  tau_Th = p.tau_Th;
+  act_flag = p.act_flag;
+  reset_flag = p.reset_flag;
+
+  // copy C-arrays
+  for ( size_t n = 0; n < NCOMP - 1; ++n )
+    g_conn[ n ] = p.g_conn[ n ];
+
+  for ( size_t n = 0; n < NCOMP; ++n )
+  {
+    t_L[ n ] = p.t_L[ n ];
+    nt_L[ n ] = p.nt_L[ n ];
+    g_L[ n ] = p.g_L[ n ];
+    C_m[ n ] = p.C_m[ n ];
+    E_ex[ n ] = p.E_ex[ n ];
+    E_in[ n ] = p.E_in[ n ];
+    E_L[ n ] = p.E_L[ n ];
+    tau_synE[ n ] = p.tau_synE[ n ];
+    tau_synI[ n ] = p.tau_synI[ n ];
+    I_e[ n ] = p.I_e[ n ];
+    // changes
+    tau_curAP[ n ] = p.tau_curAP[ n ];
+    amp_curAP[ n ] = p.amp_curAP[ n ];
+  }
+
+  return *this;
+}
+
+
+nest::iaf_cond_alpha_mc_kinetics::State_::State_( const Parameters_& p )
+  : r_( 0 )
+  ,
+  // changes
+  numCa_( 0 )
+  , th_( p.V_th )
+
+{
+  // for simplicity, we first initialize all values to 0,
+  // then set the membrane potentials for each compartment
+  for ( size_t i = 0; i < STATE_VEC_SIZE; ++i )
+    y_[ i ] = 0;
+
+  y_[ idx( 0, V_M ) ] = -70.;
+  y_[ idx( 1, V_M ) ] = -65.;
+  y_[ idx( 2, V_M ) ] = -60.;
+
+  const double_t minf =
+    1.0 / ( 1.0 + pow( numerics::e, ( ( y_[ idx( 2, V_M ) ] - p.half_m ) * -1.0 * p.slope_m ) ) );
```

Comment:
- See above about using `std::exp()`. Also, in the argument, write `-p.slope_m * ( y_ ... )`
- Don't use magic indices, use enumeration names: `y_[ idx( DIST, V_M ) ]`